botcore/_bot.py

```diff
@@ -74,7 +74,7 @@ def __init__(
 
         self.statsd_url: Optional[str] = None
         self._statsd_timerhandle: Optional[asyncio.TimerHandle] = None
-        self._guild_available = asyncio.Event()
+        self._guild_available: Optional[asyncio.Event] = None
```

Comment:
I read the commit message but I'm not sure I follow why it can't be an Event right away

Reply:
It was done since the class could be instantiated from another event loop to what calls `.start()`, so this ensures that the Event is always created within the same event loop.